### PR TITLE
Gaia button

### PR DESCRIPTION
**Pull request by guynir42**
Adds a button that queries Gaia for nearby sources and adds some of their properties as annotations.

### CODE REVIEW

**Comment by guynir42** [comment]
Addresses #2980

File: skyportal/handlers/api/annotation_services.py [hunk 1]
```diff
@@ -19,6 +23,188 @@
...
+                      the config file, or 2 arcsec if not specified in the config.
+                  crossmatchLimmag:
+                    required: false
+                    type: number
+                    description: |
+                      Crossmatch limiting magnitude (for Gaia G mag).
+                      Will ignore sources fainter than this magnitude.
+                      If not specified, will use the default value in
+                      the config file, or None if not specified in the config.
+                      If zero or None, will take sources of any magnitude.
```
Comment (mcoughlin) [inline skyportal/handlers/api/annotation_services.py:73]: 0 like false?

File: skyportal/handlers/api/annotation_services.py [hunk 2]
```diff
@@ -19,6 +23,188 @@
...
+        )
+
+        # first remove rows that have faint magnitudes
+        if limmag:  # do not remove if limmag is None or zero
+            df = df[df['phot_g_mean_mag'] < limmag]
+
+        # propagate the stars using Gaia proper motion
+        # then choose the closest match
+        if len(df) > 1:
+            df['adjusted_dist'] = 0  # new column
```
Comment (mcoughlin) [inline skyportal/handlers/api/annotation_services.py:140]: 0 vs None or nan?

File: skyportal/handlers/api/annotation_services.py [hunk 3]
```diff
@@ -19,6 +23,188 @@
...
+            for index, row in df.iterrows():
+                c = SkyCoord(
```
Comment (mcoughlin) [inline skyportal/handlers/api/annotation_services.py:142]: Nice

**Review by mcoughlin — APPROVED** [review]
@guynir42 a few missing documentation pieces looks like, but otherwise a really nice refactor.

File: skyportal/handlers/api/annotation_services.py (hunk 3 above)
Comment (guynir42) [reply skyportal/handlers/api/annotation_services.py:142]: It's nice because I took it from something Josh did for the offsets.

**Pull request merged**